modules/_createIndexFinder.js

```diff
@@ -1,28 +1,28 @@
 import getLength from './_getLength.js';
-import { slice } from './_setup.js';
+import binarySearch from './_binarySearch.js';
+import identity from './identity';
+import less from './_less.js';
+import lessEqual from './_lessEqual.js';
 import isNaN from './isNaN.js';
+import linearSearch from './_linearSearch.js';
 
-// Internal function to generate the `_.indexOf` and `_.lastIndexOf` functions.
-export default function createIndexFinder(dir, predicateFind, sortedIndex) {
-  return function(array, item, idx) {
-    var i = 0, length = getLength(array);
-    if (typeof idx == 'number') {
-      if (dir > 0) {
-        i = idx >= 0 ? idx : Math.max(idx + length, i);
-      } else {
-        length = idx >= 0 ? Math.min(idx + 1, length) : idx + length + 1;
-      }
-    } else if (sortedIndex && idx && length) {
-      idx = sortedIndex(array, item);
-      return array[idx] === item ? idx : -1;
+// Internal function to generate the `indexOf` and `lastIndexOf` functions.
+export default function createIndexFinder(dir) {
+  var forward = dir > 0;
+  var compare = forward ? less : lessEqual;
+  // `controlArg` may be either a number indicating the first index to start
+  // searching at, or a boolean indicating whether the array is sorted by native
+  // operator `<`.
+  return function(array, item, controlArg) {
+    var start;
+    if (getLength(array) < 1) return -1;
+    if (typeof controlArg == 'number') {
+      start = controlArg;
+    } else if (controlArg && forward) {
```

Comment:
The `&& forward` condition is only here to prevent the `isSorted` option from creeping into `_.lastIndexOf` as a side effect, purely because I vowed not to make any interface changes on this branch. It can be removed in a future update.